src/test/ui/lifetimes/lifetime-in-method-call.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+struct S;
+
+impl S {
+    fn foo(&self, x: &usize) {}
+}
+
+fn foo(x: &usize) {}
+
+fn main() {
+    'l: loop {
+        break 'l;
+    }
+    'y: loop {
+        let x = 3;
+        foo(&'a x);
```

Comment:
I'm confused -- what does this have to do with *method calls* in particular?

Reply:
That is, what happens if we do `let x = &'a y`?

Reply:
I called out method calls in particular because that was the mistake I saw in the wild, but yes, it should apply to anywhere that is not an ADT/method definition or a binding type.

From last nights quick test (as I was intrigued by the same):

```
error: found unexpected lifetime `'x`
 --> file.rs:2:25
  |
2 |     let x: &'x usize = &'x y;
  |                         ^^ unexpected lifetime
  |
  = help: lifetimes are only appropriate in definitions
```